src/service.js

```diff
@@ -304,11 +304,11 @@ function getAmpdocService(win) {
  * @param {!Object} holder Object holding the service instance.
  * @param {!Window|!./service/ampdoc-impl.AmpDoc} context Win or AmpDoc.
  * @param {string} id of the service.
- * @param {function(!Window):T|function(!./service/ampdoc-impl.AmpDoc):T=} opt_factory
+ * @param {function(?):T=} opt_factory
```

Comment:
Can't do union of function types in NTI also `{function(X|Y)}` won't work because `{function(X)}` is NOT `{function(X|Y)}`. This is an internal method, so not a big issue.